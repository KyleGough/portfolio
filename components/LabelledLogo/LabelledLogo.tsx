import Image from 'next/image';
import React from 'react';

interface LabelledLogoProps {
  name: string;
  logo: string;
}

export const LabelledLogo: React.FC<LabelledLogoProps> = ({ name, logo }) => (
  <div className="text-center mb-8">
    <div className="w-8 h-8 mb-2 mx-auto">
      <Image src={logo} alt={`${name} Logo`} width={30} height={30} />
    </div>
    <p className="text-sm">{name}</p>
  </div>
);
