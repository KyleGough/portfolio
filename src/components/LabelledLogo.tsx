export default function LabelledLogo(props: { name: string, logo: string }): React.ReactElement {
    return (
        <div className='text-center mb-8'>
            <img loading='lazy' className='w-8 h-8 mb-2 mx-auto' src={props.logo} alt={`${props.name} Logo`} />
            <p className='text-sm'>{props.name}</p>
        </div>
    )
}