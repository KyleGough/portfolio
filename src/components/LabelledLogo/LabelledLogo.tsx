import React from 'react';

interface LabelledLogoProps {
  name: string;
  logo: string;
}

export const LabelledLogo: React.FC<LabelledLogoProps> = ({ name, logo }) => (
  <div className="text-center mb-8">
    <img
      loading="lazy"
      className="w-8 h-8 mb-2 mx-auto"
      src={logo}
      alt={`${name} Logo`}
    />
    <p className="text-sm">{name}</p>
  </div>
);
