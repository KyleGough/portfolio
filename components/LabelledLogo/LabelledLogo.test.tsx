import { render, screen } from '@testing-library/react';
import React from 'react';

import { LabelledLogo } from './LabelledLogo';

describe('LabelledLogo component', () => {
  it('renders', () => {
    const { container } = render(<LabelledLogo name="React" logo="/#" />);
    expect(screen.getByText('React')).toBeVisible();
    expect(container).toMatchSnapshot();
  });
});
