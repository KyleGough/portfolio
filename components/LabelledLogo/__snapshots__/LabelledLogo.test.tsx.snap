// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LabelledLogo component renders 1`] = `
<div>
  <div
    class="text-center mb-8"
  >
    <div
      class="w-8 h-8 mb-2 mx-auto"
    >
      <img
        alt="React Logo"
        data-nimg="1"
        decoding="async"
        height="30"
        loading="lazy"
        src="/_next/image?url=%2F%23&w=64&q=75"
        srcset="/_next/image?url=%2F%23&w=32&q=75 1x, /_next/image?url=%2F%23&w=64&q=75 2x"
        style="color: transparent;"
        width="30"
      />
    </div>
    <p
      class="text-sm"
    >
      React
    </p>
  </div>
</div>
`;
